docs/how-to/contribute-code.rst

```diff
@@ -1,93 +1,71 @@
-============================
-Contributing code to Rubicon
-============================
+=================================
+How to contribute code to Rubicon
+=================================
 
-If you experience problems with Rubicon Java, `log them on GitHub`_. If you
-want to contribute code, please `fork the code`_ and `submit a pull request`_.
+If you experience problems with Rubicon, `log them on GitHub`_. If you want
+to contribute code, please `fork the code`_ and `submit a pull request`_.
 
 .. _log them on Github: https://github.com/beeware/rubicon-java/issues
 .. _fork the code: https://github.com/beeware/rubicon-java
 .. _submit a pull request: https://github.com/beeware/rubicon-java/pulls
 
-Setting up your development environment
----------------------------------------
+.. _setup-dev-environment:
 
-The recommended way of setting up your development environment for Rubicon is
-to use a `virtual environment <https://docs.python.org/3/library/venv.html>`__,
-install the required dependencies and start coding:
-
-.. tabs::
-
-  .. group-tab:: macOS
-
-    .. code-block:: bash
-
-      $ git clone https://github.com/beeware/rubicon-java.git
-      $ cd rubicon-java
-      $ python3 -m venv venv
-      $ . venv/bin/activate
-
-  .. group-tab:: Linux
-
-    .. code-block:: bash
-
-      $ git clone https://github.com/beeware/rubicon-java.git
-      $ cd rubicon-java
-      $ python3 -m venv venv
-      $ . venv/bin/activate
+Set up your development environment
+===================================
 
-  .. group-tab:: Windows
-
-    .. code-block:: doscon
-
-      C:\...>git clone https://github.com/beeware/rubicon-java.git
-      C:\...>cd rubicon-java
-      C:\...>py -m venv venv
-      C:\...>venv\Scripts\activate
-
-To install all the development version of Rubicon, along with all it's
-requirements, run the following commands within your virtual environment:
-
-.. tabs::
-
-  .. group-tab:: macOS
-
-    .. code-block:: bash
+The recommended way of setting up your development environment for Rubicon is
+to install a virtual environment, install the required dependencies and start
+coding:
 
-      $ (venv) pip install -e .
+.. code-block:: sh
 
-  .. group-tab:: Linux
+    $ python3 -m venv venv
+    $ source venv/bin/activate.sh
+    (venv) $ pip install --upgrade pip
+    (venv) $ pip install --upgrade setuptools
+    (venv) $ pip install tox
+    (venv) $ git clone https://github.com/beeware/rubicon-java.git
+    (venv) $ cd rubicon-objc
+    (venv) $ pip install -e .
 
-    .. code-block:: bash
+Next, check if `python-config` passes exists in your environment, and points at
+your current active Python 3 install. You can confirm this by running:
 
-      $ (venv) pip install -e .
+.. code-block:: sh
 
-  .. group-tab:: Windows
+   (venv) $ python-config --prefix
```

Comment:
Same non-showstopper note about `python3-config`.

Reply:
Dumb question: if `python-config --prefix` works, does that print the path to the virtualenv? Or to the the actual Python interpreter?

If the latter, how would a user know they got it right?

Reply:
Not sure I have a good answer here. 

However, further testing on my own setup reveals that `python3-config` *always* returns the python3.7 config, so it looks like setting `PYTHON_CONFIG` is going to be good advice regardless.

To that end - is there any reason why shouldn't just evaluate `PYTHON_CONFIG` in the Makefile? If it can be reliably determined from the resolved location of the python executable (which it looks like it can be), this makes this who documentation question moot, and makes the testing setup even easier. I'll try this in a followon PR.

Reply:
Turns out it works, which lets us simplify the CircleCI config and setup instructions even more.